docs/design-docs/modular-cli/middleware-modules.md

```diff
@@ -0,0 +1,40 @@
+# Ansible based module
+
+## Purpose
+
+To provide separation of concern on middleware level code we need to have consistent way to produce ansible based modules.
+
+## Requirements
+
+There are following requirements for modules:
+ * Allow two-ways communication with other modules via Statefile
+ * Allow a reuse of ansible roles between modules
+
+## Design
+
+### Components
+1. Docker – infrastructure modules are created as Docker containers so far so this approach should continue
+2. Ansible – we do have tons of ansible code which could be potentially reused. Ansible is also a _de facto_ industry standard for software provisioning, configuration management, and application deployment.
+3. Ansible-runner – do to need of automation we should use ansible-runner application which is a wrapper for ansible commands (i.e.: ansible-playbook) and provides good code level integration features (i.e.: passing of variables to playbook, extracting logs, RC and facts cache). It is originally used in AWX.
```

Comment:
`do to need of automation` - is that correct?

Reply:
Typo.